test/router.spec.ats

```diff
@@ -167,13 +167,11 @@ describe('RootRouter', () => {
 
       it('should generate URLs', sync(async () => {
         expect(child.generate('C')).toBe('/a/c');
-        expect(router.generate('C')).toBe('/a/c');
```

Comment:
This is a consequence of the "recognizer chain" change - we're not exploring deep into possible child states any more.

Question - is that actually the desired behaviour? That we can get to any possible child URL by making routes up as we go from anywhere else in the tree, as long as there's a path? It seems like this might cause some unpredictability in routing, i.e.:

In /settings, where `settings` component defines a child /welcome route with component `welcome` and a recursive /settings with another embedded `settings` component. Should settingsRouter.generate('welcome') link to /welcome or /settings/welcome or even /settings/settings/welcome?

It seems like being unpredictable is not something we'd want in a router, so this path search should go. Maybe introduce a DSL for getting to remote subcomponents? Something like `router.generate('settings:settings:welcome') === '/settings/settings/welcome'`?